hub/herald/session.py

```diff
@@ -282,8 +284,19 @@ async def _start_external_servers(self):
         """
         env = self.env
         host = env.cs_host()
-        if env.tcp_port is not None:
-            await self._start_server('TCP', host, env.tcp_port)
+        if env.tcp_port is None:
+            return
+        started = False
+        while not started:
+            try:
+                await self._start_server('TCP', host, env.tcp_port)
+                started = True
+            except OSError as e:
+                if e.errno is errno.EADDRINUSE:
+                    await asyncio.sleep(3)
```

Comment:
Can this log as a warning? Otherwise LGTM

Reply:
Are there other socket errors we might get?

Reply:
`_start_server()` logs at error level already.

`run_es_notifier()` looks like it would be vulnerable to the same thing. I have not seen it in the test failures, but that could be because tests don't stop/start elastic sync a lot.

I haven't been able to determine whether there are other transient error codes.

Reply:
Fixed `run_es_notifier()` in similar way. Added broad logging of Exception types.